### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by arjun-computer-geek**
Authentication completed.

check Login Feature
Check Protected Routes
Check wish list

### CODE REVIEW

File: frontend/src/context/userContext.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ const UserProvider = ({ children }) => {
   // use reducer hooks
   const [state, dispatch] = useReducer(authReducer, {
```
Comment (Shubhamkr2610) [inline frontend/src/context/userContext.js]: appreciation: its good to give message

**Review by Shubhamkr2610 — APPROVED** [review]

File: frontend/src/components/route/ProtectedRoute.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const ProtectedRoute = () => {
+    const {
+      state: { isAuthenticated },
+    } = useUser();
+
+
+  if (!isAuthenticated) return <Navigate to="/login" replace />;
+  return <Outlet />;
+};
```
Comment (Shubhamkr2610) [inline frontend/src/components/route/ProtectedRoute.js:14]: Your logic is good . Easy to understand

**Review by Shubhamkr2610 — APPROVED** [review]

**Review by Shubhamkr2610 — APPROVED** [review]

File: frontend/src/actions/userActions.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  LOGIN_FAIL,
+  REGISTER_USER_REQUEST,
+  REGISTER_USER_SUCCESS,
+  REGISTER_USER_FAIL,
+  LOAD_USER_REQUEST,
+  LOAD_USER_SUCCESS,
+  LOAD_USER_FAIL,
+  LOGOUT_SUCCESS,
+  LOGOUT_FAIL,
+} from "../constants/userConstants";
```
Comment (sanjitsarkar) [inline frontend/src/actions/userActions.js:15]: Proper use of CONSTANTS

**Review by sanjitsarkar — APPROVED** [review]

**Comment by sanjitsarkar** [comment]
Everything looks good and I liked the practice of using frequently used names with Constants

**Pull request merged**